test/e2e/runner/docker.go

```diff
@@ -0,0 +1,219 @@
+// nolint: gosec
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"text/template"
+
+	"github.com/tendermint/tendermint/libs/log"
+	e2e "github.com/tendermint/tendermint/test/e2e/pkg"
+)
+
+// DockerInfra provides an API for provisioning and manipulating infrastructure
+// for Docker-based testnets.
+type DockerInfra struct {
+	logger  log.Logger
+	testnet *e2e.Testnet
+}
+
+var _ Infra = &DockerInfra{}
+
+// NewDockerInfra constructs an infrastructure provider that allows for
+// Docker-based testnet infrastructure.
+func NewDockerInfra(logger log.Logger, testnet *e2e.Testnet) *DockerInfra {
+	return &DockerInfra{
+		logger:  logger,
+		testnet: testnet,
+	}
+}
+
+func (i *DockerInfra) GenerateConfig() error {
```

Comment:
isn't this an implementation detail of some other operation?

Reply:
I've renamed this now to `Setup` to be more generic.